packages/kbn-securitysolution-grouping/src/components/group_selector/index.tsx

```diff
@@ -43,8 +43,17 @@ const GroupSelectorComponent = ({
     [groupsSelected]
   );
 
-  const panels: EuiContextMenuPanelDescriptor[] = useMemo(
-    () => [
+  const panels: EuiContextMenuPanelDescriptor[] = useMemo(() => {
+    const isOptionDisabled = (key?: string) => {
+      // Do not disable when maxGroupingLevels is 1 to allow toggling between groups
+      if (maxGroupingLevels === 1) {
+        return false;
+      }
+      // Disable all non selected options when the maxGroupingLevels is reached
+      return groupsSelected.length === maxGroupingLevels && (key ? !isGroupSelected(key) : true);
```

Comment:
should we keep the behaviour of disabling currently selected option when the max level is 1? Right now to remove grouping, you can either switch to none or "unselect" the option. If we change the behavior to toggle when max level is 1, maybe it makes sense to disable this "unselect" behaviour

Reply:
great point, I'll adjust it in the code and bring it to discussion with the code owners

Reply:
I added the unit tests related to the changes introduced in this PR, thanks a lot for reviewing it @stephmilovic!